### PR TITLE
Fix design issues with sensitive preview cards

### PR DESCRIPTION
**Pull request by ClearlyClaire**
## Before

The “sensitive” button is centered on the image part, which is somewhat sensible for media cards, but not for cards which only have a small image to their side.

Furthermore, clicking outside that image area would not do anything in the case of a media card, and clicking *anywhere* would also trigger the link in the other case.

image

## After

The “sensitive” button is changed to cover the whole preview card, regardless of the preview type.
Clicking a sensitive preview card anywhere now reveals it, without opening the link behind it.

image

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This looks worse than before for me personally. I would like to see this reverted.

**Comment by ThibG** [comment]
@SuperSandro2000 changed that part back to centering on the preview image for interactive cards

**Comment by Gargron** [comment]
My wishlist is:

- Don't blur text, only thumbnail
- Don't do anything if there is no thumbnail (e.g. don't put overlay over the "document" icon)
- If it's in the small format, don't bother with "sensitive content" label and unhiding

**Comment by ThibG** [comment]
hmmm, I kind of disagree with that, the text could very well contain sensitive content one would not want to see without clicking first, it's not just graphic imagery.

Likewise, the thumbnail for the “small format” may also contain graphic imagery that one would not want to see directly.

**Comment by Gargron** [comment]
> Likewise, the thumbnail for the “small format” may also contain graphic imagery that one would not want to see directly.

I'm not saying to not blur it, I'm saying to not bother with controlling it. Just leave it blurred.

**Comment by ThibG** [comment]
That sounds a little more sensible, but hm, I still feel like the text could be sensitive and you'd want it not directly readable…

File: app/javascript/mastodon/features/status/components/card.js [hunk 1]
```diff
@@ -164,8 +164,10 @@ export default class Card extends React.PureComponent {
     this.setState({ previewLoaded: true });
   }
 
-  handleReveal = () => {
+  handleReveal = e => {
     this.setState({ revealed: true });
+    e.preventDefault();
```
Comment (Gargron) [inline app/javascript/mastodon/features/status/components/card.js:169]: Feel like this should be the first thing in the method

File: app/javascript/mastodon/features/status/components/card.js [hunk 2]
```diff
@@ -252,7 +254,7 @@ export default class Card extends React.PureComponent {
       }
 
       return (
-        <div className={className} ref={this.setRef}>
+        <div className={className} ref={this.setRef} onClick={this.handleReveal}>
```
Comment (Gargron) [inline app/javascript/mastodon/features/status/components/card.js:257]: Needs a `role='button'` or perhaps `role={sensitive ? 'button' : null}`

**Review by Gargron — APPROVED** [review]

**Pull request merged**